### ISSUE TITLE
Provide relevant warning message in cases where maxLength is present on non-string types

### ISSUE THREAD

**Issue by mohit-s96** [RESOStandards/reso-certification-utils#91]
This is an issue that leads to the generation of an invalid schema which causes AJV to crash when compiling in strict mode.

For eg.
```json
{
  "resourceName": "Property",
  "fieldName": "InternetConsumerCommentYN",
  "nullable": true,
  "annotations": [],
  "type": "Edm.Boolean",
  "maxLength": 1
}
```
Here the use of `maxLength` is non-standard (refer: [odata spec](https://docs.oasis-open.org/odata/odata/v4.01/csprd01/part3-csdl/odata-v4.01-csprd01-part3-csdl.html#_Toc469935983)).

We can catch this issue using a small check in the validation step. Since this issue isn't related to the payload being validated, we may also want to give this kind of warning a new type (like 'SCHEMA_ERROR', or 'METADATA_ERROR' etc) so that the frontend can distinguish this warning from normal errors/warnings.

__Structure:__ (please suggest changes)
```json
{
  "description": "RESO Common Format Schema Validation Summary",
  "generatedOn": "2023-12-19T16:54:55.246Z",
  "totalErrors": 0,
  "totalWarnings": 1,
  "items": [
    {
      "type": "SCHEMA_ERROR",
      "warnings": [
        {
          "message": "maxLength keyword found on type number",
          "link": "https://docs.oasis-open.org/odata/odata/v4.01/csprd01/part3-csdl/odata-v4.01-csprd01-part3-csdl.html#_Toc469935983",
          "keyword": "maxLength",
          "occurrences": [
            {
              "resourceName": "Lookup",
              "fieldName": "InternetConsumerCommentYN"
            }
          ]
        }
      ]
    }
  ]
}
```

cc: @darnjo

**Comment by darnjo** [comment]
Closed by #92

**Comment by darnjo** [comment]
Closed by #92
